app/models/disability_claim_document.rb

```diff
@@ -47,6 +47,14 @@ def ==(other)
     attributes == other.attributes
   end
 
+  # Redefining attributes method so that it picks up changes made after initialization
+  def attributes
+    Hash[attribute_set.map { |attr| [attr.name.to_s, send(attr.name)] }]
+  end
+
+  alias to_h attributes
+  alias to_hash attributes
```

Comment:
I originally tried it without these lines and the tests failed. I suspect that it ends up aliasing the method on the superclass instead of the subclass if you leave them out.